File: Android/aiconversationkit/src/main/java/com/trtc/uikit/aiconversationkit/manager/internal/PackageService.java
package com.trtc.uikit.aiconversationkit.manager.internal;

import android.text.TextUtils;

import com.trtc.tuikit.common.system.ContextProvider;

public class PackageService {
    private static final String PACKAGE_RT_CUBE     = "com.tencent.trtc";
    private static final String PACKAGE_TENCENT_RTC = "com.tencent.rtc.app";

    public static boolean isInternalDemo() {
        return isRTCube() || isTencentRTC();
    }

    public static boolean isRTCube() {
        return TextUtils.equals(PACKAGE_RT_CUBE, ContextProvider.getApplicationContext().getPackageName());
    }

    public static boolean isTencentRTC() {
        return TextUtils.equals(PACKAGE_TENCENT_RTC, ContextProvider.getApplicationContext().getPackageName());
    }
}

File: iOS/AIConversationKit/Manager/PackageService.swift
//
//  PackageService.swift
//  AIConversationKit
//
//  Created on 2026/2/26.
//

import Foundation

enum PackageService {
    private static let rtCubeBundleId = "com.tencent.mrtc"
    private static let tencentRTCBundleId = "com.tencent.rtc.app"

    static var isInternalDemo: Bool {
        isRTCube || isTencentRTC
    }

    static var isRTCube: Bool {
        Bundle.main.bundleIdentifier == rtCubeBundleId
    }

    static var isTencentRTC: Bool {
        Bundle.main.bundleIdentifier == tencentRTCBundleId
    }
}
